library/s3/index.js

```diff
@@ -220,32 +220,34 @@ function s3Wrapper(credentialsOpts) {
       const {
         Versions: keyVersionObjects, IsTruncated, NextKeyMarker, VersionIdMarker,
       } = await s3.listObjectVersions(listVersionParams).promise();
-      if (IsTruncated) {
-        const keyVersionArrays = await fetchTruncatedVersionS3Files({
-          params: listVersionParams,
-          keyObjects: keyVersionObjects,
-          NextKeyMarker,
-          ...VersionIdMarker && { VersionIdMarker },
-        });
-        let counter = 0;
+      if (keyVersionObjects.length > 0) {
```

Comment:
When you're deleting, just like when you're doing a GET on a resource by ID, you typically assume the resource exists (because you know the ID). Consider logging out a message here when the file doesn't exist. This would be the equivalent of returning a 404 Not Found.

Reply:
done